nfsv4/client.c

```diff
@@ -110,32 +114,39 @@ int nfs4_append(nfs4_file f, void *source, bytes length)
     p.mask = NFS4_PROP_SIZE;
     p.size = f->expected_size;
     // get the offset in case we fail
-    push_op(r, OP_GETATTR, set_expected_size, &f);    
+    push_op(r, OP_GETATTR, set_expected_size, &f);
     push_fattr_mask(r, NFS4_PROP_SIZE);
     push_op(r, OP_VERIFY, 0, 0);
     push_fattr(r, &p);
     push_op(r, OP_SETATTR, parse_attrmask, 0);
-    push_stateid(r, &f->open_sid);    
+    push_stateid(r, &f->open_sid);
     push_fattr(r, &p);    
     push_lock(r, &f->open_sid, WRITE_LT, f->expected_size, f->expected_size + length, &f->latest_sid);
     buffer b = alloca_wrap_buffer(source, length);
     u64 offset = f->expected_size;
-        
+ 
+    transact(r);
```

Comment:
Most of the time in these `nfs4_smtg` functions, `transact(r)` is wrapped in `api_check(c, transact(r))` to stop and return upon failure. Are there steps here that need to execute even in the case of failure?

Reply:
Also for my understanding, why is this a transact rather than a rpc_send like on line 134

Reply:
> Also for my understanding, why is this a transact rather than a rpc_send like on line 134

rpc_send will have file busy error, I think it is because the client needs to get the lock before it starts writing to the file.

Reply:
> Most of the time in these `nfs4_smtg` functions, `transact(r)` is wrapped in `api_check(c, transact(r))` to stop and return upon failure. Are there steps here that need to execute even in the case of failure?

good catch, I will add the error checking